perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -1918,14 +1920,17 @@ public User convertUserEmptyStringsInObjectAttributesIntoNull(User user) {
 	}
 
 	@Override
-	public void changeNonAuthzPassword(PerunSession sess, User user, String m, String password, String lang) throws LoginNotExistsException, PasswordChangeFailedException, PasswordOperationTimeoutException, PasswordStrengthFailedException, InvalidLoginException, PasswordStrengthException {
+	public void checkPasswordResetRequestIsValid(PerunSession sess, User user, String m) throws PasswordResetLinkExpiredException, PasswordResetLinkNotValidException {
+		int requestId = Integer.parseInt(Utils.cipherInput(m, true));
 
-		String requestId = Utils.cipherInput(m, true);
-		Pair<String,String> resetRequest = getUsersManagerImpl().loadPasswordResetRequest(user, Integer.parseInt(requestId));
+		usersManagerImpl.checkPasswordResetRequestIsValid(sess, user, requestId);
```

Comment:
You should use ``getUsersManagerImpl()`` method, so that Spring will use its proxy class and respect transaction config. It would probably worked here, since you only read, but we should be consistent in getting manager classes.